src/Dynatrace.OpenTelemetry.Exporter.Metrics.Tests/DynatraceMetricsExporterTests.cs

```diff
@@ -334,98 +286,27 @@ public void Export_WithTooLargeMetric_ShouldNotSendRequest()
-			var mockMessageHandler = SetupHttpMock((HttpRequestMessage r) => actualRequestMessage = r);
-
-			var sut = new DynatraceMetricsExporter(null, null, new HttpClient(mockMessageHandler.Object));
-
-			var exporterProxy = new TestMetricsExporterProxy(sut);
-
-			var metricReader = new BaseExportingMetricReader(exporterProxy);
-			using var provider = Sdk.CreateMeterProviderBuilder()
-				.AddMeter(meter.Name)
-				.AddReader(metricReader)
-				// configure another reader with Cumulative aggregation
-				.AddConsoleExporter(opt => opt.AggregationTemporality = AggregationTemporality.Cumulative)
-				.Build();
-
-			var counter = meter.CreateCounter<int>("counter");
-
-			for (var i = 10; i <= 30; i += 10)
-			{
-				counter.Add(i, _attributes);
-
-				metricReader.Collect();
-
-				// Assert
-				var exportedMetrics = exporterProxy.GetExportedMetrics();
-				var point = MetricTest.FromMetricPoints(exportedMetrics.First().GetMetricPoints()).First();
-
-				var expected = $"counter,attr1=v1,attr2=v2,dt.metrics.source=opentelemetry count,delta={i} {point.TimeStamp}";
-				var actualMetricString = await actualRequestMessage.Content!.ReadAsStringAsync();
-				Assert.Equal(expected, actualMetricString);
-
-				AssertExportRequest(actualRequestMessage);
-			}
-
-			mockMessageHandler.Protected().Verify(
-				"SendAsync",
-				Times.Exactly(3),
-				ItExpr.IsAny<HttpRequestMessage>(),
-				ItExpr.IsAny<CancellationToken>());
-		}
-
 		[Fact]
 		public async Task Export_SimulateMultipleExports_ShouldExportCorrectDelta()
 		{
 			// Arrange
-			using var meter = new Meter(TestUtils.GetCurrentMethodName(), "0.0.1");
-
 			HttpRequestMessage actualRequestMessage = null!;
-			var mockMessageHandler = SetupHttpMock((HttpRequestMessage r) => actualRequestMessage = r);
+			var mockMessageHandler = SetupHttpMock(r => actualRequestMessage = r);
 
 			var sut = new DynatraceMetricsExporter(null, null, new HttpClient(mockMessageHandler.Object));
 
-			var exporterProxy = new TestMetricsExporterProxy(sut);
-
-			var metricReader = new BaseExportingMetricReader(exporterProxy);
-			using var provider = Sdk.CreateMeterProviderBuilder()
-				.AddMeter(meter.Name)
-				.AddReader(metricReader)
-				.Build();
-
-			var counter = meter.CreateCounter<int>("counter");
+			var counter = _meter.CreateCounter<int>("counter");
 
 			for (var i = 10; i <= 30; i += 10)
 			{
 				counter.Add(i, _attributes);
 
-				metricReader.Collect();
+				_exportedMetrics.Clear();
+				_meterProvider.ForceFlush();
+				sut.Export(new Batch<Metric>(_exportedMetrics.ToArray(), _exportedMetrics.Count));
 
 				// Assert
-				var exportedMetrics = exporterProxy.GetExportedMetrics();
-				var point = MetricTest.FromMetricPoints(exportedMetrics.First().GetMetricPoints()).First();
+				var point = MetricTest.FromMetricPoints(_exportedMetrics.First().GetMetricPoints()).First();
 
 				var expected = $"counter,attr1=v1,attr2=v2,dt.metrics.source=opentelemetry count,delta={i} {point.TimeStamp}";
```

Comment:
Should this not export 10 on each export? or am I misunderstanding the test?

Reply:
counter adds `10` in the first loop, `20` in the second loop and `30` in the third loop and flushes every loop. Therefore deltas will be `10`, `20`, and `30` :slightly_smiling_face:

Reply:
💡